test/cypress/integration/simplified-account/service-settings/switch-psp/switch-to-worldpay/add-worldpay-credential.cy.js

```diff
@@ -0,0 +1,237 @@
+const VALID_WORLDPAY_PASSWORD = WORLDPAY_CREDENTIAL_IN_ENTERED_STATE.credentials.one_off_customer_initiated.password
+const SWITCH_TO_WORLDPAY_SETTINGS_URL = `/simplified/service/${SERVICE_EXTERNAL_ID}/account/${LIVE_ACCOUNT_TYPE}/settings/switch-psp/switch-to-worldpay`
+const SWITCH_TO_WORLDPAY_ADD_CREDENTIAL_TASK_SETTINGS_URL = `/simplified/service/${SERVICE_EXTERNAL_ID}/account/${LIVE_ACCOUNT_TYPE}/settings/switch-psp/switch-to-worldpay/worldpay-details/one-off-customer-initiated`
+// ---
+
+const setStubs = (opts = {}, additionalStubs = []) => {
+  const pendingCredential = opts.pendingCredential || WORLDPAY_CREDENTIAL_IN_CREATED_STATE
+  cy.task('setupStubs', [
+    userStubs.getUserSuccess({
+      userExternalId: USER_EXTERNAL_ID,
+      gatewayAccountId: GATEWAY_ACCOUNT_ID,
+      serviceName: SERVICE_NAME,
+      serviceExternalId: SERVICE_EXTERNAL_ID,
+      role: ROLES[opts.role || 'admin'],
+      features: 'degatewayaccountification' // TODO remove features once simplified accounts are live
+    }),
+    gatewayAccountStubs.getAccountByServiceIdAndAccountType(SERVICE_EXTERNAL_ID, LIVE_ACCOUNT_TYPE, {
+      gateway_account_id: GATEWAY_ACCOUNT_ID,
+      type: LIVE_ACCOUNT_TYPE,
+      payment_provider: STRIPE,
+      provider_switch_enabled: true,
+      allow_moto: opts.moto || false,
+      gateway_account_credentials: [
+        STRIPE_CREDENTIAL_IN_ACTIVE_STATE,
+        pendingCredential
+      ]
+    }),
+    ...additionalStubs])
+}
+
+describe('Add Worldpay credential task', () => {
+  beforeEach(() => {
+    cy.setEncryptedCookies(USER_EXTERNAL_ID)
+  })
+  describe('The settings nav', () => {
+    beforeEach(() => {
+      setStubs({}, [])
+      cy.visit(SWITCH_TO_WORLDPAY_ADD_CREDENTIAL_TASK_SETTINGS_URL)
+    })
+    checkSettingsNavigation('Switch to Worldpay', SWITCH_TO_WORLDPAY_SETTINGS_URL)
+  })
+  describe('The page', () => {
+    beforeEach(() => {
+      setStubs({}, [])
+      cy.visit(SWITCH_TO_WORLDPAY_ADD_CREDENTIAL_TASK_SETTINGS_URL)
+    })
+    checkTitleAndHeading('Your Worldpay credentials', SERVICE_NAME.en)
+  })
+  describe('For a non-admin', () => {
+    beforeEach(() => {
+      setStubs({
+        role: 'view-and-refund',
+        moto: true
+      }, [])
+      cy.visit(SWITCH_TO_WORLDPAY_ADD_CREDENTIAL_TASK_SETTINGS_URL, { failOnStatusCode: false })
+    })
+    it('should show admin only error', () => {
+      cy.title().should('eq', 'An error occurred - GOV.UK Pay')
+      cy.get('h1').should('contain.text', 'An error occurred')
+      cy.get('#errorMsg').should('contain.text', 'You do not have the administrator rights to perform this operation.')
+    })
+  })
+
+  describe('For an admin', () => {
+    describe('For a MOTO service', () => {
+      describe('When no credentials have been entered previously', () => {
+        beforeEach(() => {
+          setStubs({
+            moto: true
+          })
+          cy.visit(SWITCH_TO_WORLDPAY_ADD_CREDENTIAL_TASK_SETTINGS_URL)
```

Comment:
Think it would be useful to test we can navigate to this page from the "Switch to Worldpay" screen too.

Reply:
I agree! There's already something in the `switch-to-worldpay` cypress test that checks that all tasks are navigable from the tasks index